rs/tests/consensus/tecdsa/tecdsa_add_nodes_test.rs

```diff
@@ -1,13 +1,233 @@
+/* tag::catalog[]
+Title:: Adding nodes to a subnet running canister threshold signing
+
+Goal:: Test whether removing subnet nodes impacts the threshold signature feature
+
+Runbook::
+. Setup:
+    . System subnet comprising N nodes, necessary NNS canisters, and with chain key feature featured.
+    . X unassigned nodes.
+. Enable chain key signing.
+. Get public keys for all supported schemes.
+. Add all X unassigned nodes to System subnet via proposal.
+. Assert that node membership has changed.
+. Assert that the key was reshared due to the new membership
+. Assert that all nodes are making progress
+. Assert that chain key signing continues to work with the same public key as before.
+
+Success::
+. System subnet contains N + X nodes.
+. Chain key signature succeeds with the same public key as before.
+
+end::catalog[] */
+
 use anyhow::Result;
 
+use canister_test::Canister;
+use ic_consensus_system_test_utils::rw_message::cert_state_makes_progress_with_retries;
+use ic_consensus_threshold_sig_system_test_utils::{
+    enable_chain_key_signing, get_public_key_and_test_signature, get_public_key_with_logger,
+    make_key_ids_for_all_schemes, DKG_INTERVAL,
+};
+use ic_management_canister_types::MasterPublicKeyId;
+use ic_nns_constants::GOVERNANCE_CANISTER_ID;
+use ic_nns_governance::pb::v1::NnsFunction;
+use ic_registry_subnet_type::SubnetType;
 use ic_system_test_driver::driver::group::SystemTestGroup;
+use ic_system_test_driver::driver::ic::{InternetComputer, Subnet};
+use ic_system_test_driver::driver::test_env::TestEnv;
+use ic_system_test_driver::driver::test_env_api::{
+    secs, HasPublicApiUrl, HasTopologySnapshot, IcNodeContainer, NnsInstallationBuilder,
+    SubnetSnapshot,
+};
 use ic_system_test_driver::systest;
-use ic_tests::tecdsa;
+use ic_system_test_driver::{
+    nns::{submit_external_proposal_with_test_id, vote_execute_proposal_assert_executed},
+    util::*,
+};
+use ic_types::Height;
+use registry_canister::mutations::do_add_nodes_to_subnet::AddNodesToSubnetPayload;
+use slog::{info, Logger};
+use std::collections::BTreeMap;
+
+const NODES_COUNT: usize = 4;
+const UNASSIGNED_NODES_COUNT: usize = 3;
+
+const ECDSA_KEY_TRANSCRIPT_CREATED: &str = "consensus_ecdsa_key_transcript_created";
+
+fn config(env: TestEnv) {
```

Comment:
Consider renaming this from the old `config` naming convention to the modern:
```suggestion
fn setup(env: TestEnv) {
```